File: Commission.java
import java.util.Scanner;

public class Commission{

    public static void main(String[] args) 
{
        Scanner scanner = new Scanner(System.in);

        
        System.out.print("Enter the total sales amount: ");
        double salesAmount = scanner.nextDouble();

        
        System.out.print("Enter the commission rate (in percentage): ");
        double commissionRate = scanner.nextDouble();

        
        double commission = calculateCommission(salesAmount, commissionRate);

        
        System.out.printf("The Total commission Is:"+commission);

       
    }

    
   	 public static double calculateCommission(double salesAmount, double commissionRate)
 	{
    	    return (salesAmount * commissionRate) / 100;
    	}
}

File: commission.py
class CommissionCalculator:
    def __init__(self, commission_rate):
        self.commission_rate = commission_rate

    def calculate_commission(self, sales_amount):
        if sales_amount < 0:
            raise ValueError("Sales amount cannot be negative.")
        return sales_amount * self.commission_rate

def get_user_input(prompt):
    while True:
        try:
            value = float(input(prompt))
            return value
        except ValueError:
            print("Invalid input. Please enter a numeric value.")

def main():
    commission_rate = 0.1  # 10%
    calculator = CommissionCalculator(commission_rate)

    # Get two sales amounts from the user
    sales_amount1 = get_user_input("Enter sales amount 1: ")
    sales_amount2 = get_user_input("Enter sales amount 2: ")

    # Calculate and display commissions
    for sales in [sales_amount1, sales_amount2]:
        try:
            commission = calculator.calculate_commission(sales)
            print(f"Sales Amount: ${sales:.2f} | Commission: ${commission:.2f}")
        except ValueError as e:
            print(f"Error: {e}")

if __name__ == "__main__":
    main()
